mssqlcli/mssql_cli.py

```diff
@@ -212,9 +209,14 @@ def __init__(self, options):
                              "specified.")
 
     def __del__(self):
-        # Shut-down sqltoolsservice
-        if self.sqltoolsclient:
-            self.sqltoolsclient.shutdown()
+        """ Shut down sqltoolsservice if defined. """
+        try:
+            # try/except block needed if shutdown is invoked before sqltoolsclient is defined.
+            # example: invalid row-limit argument is passed.
+            if self.sqltoolsclient:
```

Comment:
This makes me nervous. In what situations will we raise an AttributeError?

Reply:
`AttributeError` is raised f the `MssqlCli` constructor exits due to invalid parameters. In this case, say we input a series of characters instead of an integer for `--row-limit`—we exit during class instantiation, and before `sqltoolsclient` gets instantiated. Therefore the class doesn't possess the `sqltoolsclient` attribute, thus raising an `AttributeError`.